### PR TITLE
refactor: Change initialization of MongoDB objects 

### PR DESCRIPTION
**Pull request by jfonseca-aneo**
As a result of the process to to find a fix for https://github.com/aneoconsulting/ArmoniK/issues/867, this PR refactors the initialization of the mongodb objects to add better health checks, retries and more informative exception messages.

Fix https://github.com/aneoconsulting/ArmoniK/issues/921

### CODE REVIEW

File: Adaptors/MongoDB/src/ObjectStorage.cs [hunk 1]
```diff
@@ -161,6 +161,10 @@ await foreach (var chunk in objectCollection.AsQueryable(sessionHandle)
   public async IAsyncEnumerable<string> ListKeysAsync([EnumeratorCancellation] CancellationToken cancellationToken = default)
   {
     using var _                = logger_.LogFunction();
+    await sessionProvider_.Init(cancellationToken)
```
Comment (aneojgurhem) [inline Adaptors/MongoDB/src/ObjectStorage.cs:164]: I'm not sure this is the right place

Comment (jfonseca-aneo) [reply Adaptors/MongoDB/src/ObjectStorage.cs:164]: Indeed, it is not :)

File: Adaptors/MongoDB/src/Common/MongoCollectionProvider.cs [hunk 2]
```diff
@@ -1,77 +1,177 @@
...
+    Initialization = InitializeAsync(options,
+                                     sessionProvider,
+                                     mongoDatabase,
+                                     logger,
+                                     cancellationToken);
+  }
+
+  public Task<IMongoCollection<TData>> Initialization { get; private set; }
+
+  public Task<HealthCheckResult> Check(HealthCheckTag tag)
```
Comment (aneojgurhem) [inline Adaptors/MongoDB/src/Common/MongoCollectionProvider.cs:64]: missing doc or `/// <inheritdoc/ >`

File: Adaptors/MongoDB/src/Options/MongoDB.cs [hunk 3]
```diff
@@ -48,6 +48,8 @@ public class MongoDB
 
   public string Password { get; set; } = "";
 
+  public int maxRetries { get; set; } = 5;
```
Comment (aneojgurhem) [inline Adaptors/MongoDB/src/Options/MongoDB.cs:51]:
```suggestion
  public int MaxRetries { get; set; } = 5;
```

**Review by aneojgurhem — APPROVED** [review]

**Pull request merged**